backend/src/main/kotlin/blue/mild/covid/vaxx/service/IsinRetryService.kt

```diff
@@ -0,0 +1,178 @@
+import blue.mild.covid.vaxx.dto.response.IsinJobDtoOut
+import blue.mild.covid.vaxx.dto.response.PatientDtoOut
+import blue.mild.covid.vaxx.dto.response.toPatientVaccinationDetailDto
+import mu.KLogging
+
+
+class IsinRetryService(
+    private val patientService: PatientService,
+    private val patientValidation: PatientValidationService,
+    private val isinService: IsinServiceInterface,
+    private val dataCorrectnessService: DataCorrectnessService,
+    private val patientRepository: PatientRepository,
+    private val vaccinationService: VaccinationService,
+) {
+
+    private companion object : KLogging()
+
+    @Suppress("ComplexCondition", "ComplexMethod", "LongMethod") // this is complex job, it is ok here
+    suspend fun runIsinRetry(isinJobDto: IsinJobDtoIn): IsinJobDtoOut {
+        var validatedPatientsSuccess: Int = 0
+        var validatedPatientsErrors: Int = 0
+
+        var exportedPatientsInfoSuccess: Int = 0
+        var exportedPatientsInfoErrors: Int = 0
+
+        var exportedVaccinationsSuccess: Int = 0
+        var exportedVaccinationsErrors: Int = 0
+
+        val patients = patientService.getPatientsByConjunctionOf(
+            n = isinJobDto.patientsCount,
+            offset = isinJobDto.patientsOffset.toLong()
+        )
+
+        logger.debug("${patients.count()} patients will be processed.")
+
+        for (patient in patients) {
+            logger.debug("Checking ISIN id of patient ${patient.id}")
+
+            // 1. If patient has personal number but ISIN id is not set -> try ISIN validation
+            val updatedPatient = if (!patient.isinId.isNullOrBlank()) {
+                logger.info("ISIN id of patient ${patient.id} is already set, skipping ISIN validation.")
+                patient
+            } else if (isinJobDto.validatePatients && !patient.personalNumber.isNullOrBlank() ) {
+                logger.info("Patient ${patient.id} has personal number but no ISIN id. Validating in ISIN.")
+
+                val newIsinPatientId = retryPatientValidation(patient)
+
+                logger.info(
+                    "Patient ${patient.id} was validated in ISIN with " +
+                    "obtained ISIN id: ${newIsinPatientId}."
+                )
+
+                if (newIsinPatientId != null) {
+                    validatedPatientsSuccess++;
+                } else {
+                    validatedPatientsErrors++;
+                }
+                patient.copy(isinId = newIsinPatientId)
+            } else {
+                logger.info("ISIN id of patient ${patient.id} is not set but validatePatients=false. Skipping ISIN validation.")
+                patient.copy(isinId = null)
+            }
+
+            if (updatedPatient.isinId.isNullOrBlank()) continue
+
+            // 2. If data are correct but not exported to ISIN -> try export to isin
+            logger.debug("Checking correctness exported to ISIN of patient ${updatedPatient.id}")
```

Comment:
nevim jestli bych to nedal do info - obecne to logovani. Nebo na produkci muzem zapnout debug

Reply:
debug zapnout nechceme, pak uz toho leze fakt hodne ze vsech dalsich komponenet

Reply:
tak to davejme spis do info, at fakt pripadne hned vime, co opravit